File: UNIT 3/py/nilaiMatkulChecker.py
import time

print("---Nilai Matkul Checker---")
ans = "Y"
while ans == "Y" or ans == "y":
    nama = input("Masukan nama\t: ")
    nim = input("Masukan NIM\t: ")
    matkul = input("Masukan mata kuliah\t: ")
    nilai = int(input("Masukan nilai\t: "))
    print("\nData berhasil diinput!")
   
    if 85 <= nilai <= 100:
        nilaihruf = 'A'
    elif 75 <= nilai < 85:
        nilaihruf = 'B'
    elif 65 <= nilai < 75:
        nilaihruf = 'C'
    elif nilai < 65:
        nilaihruf = 'D'
    else:
        nilaihruf = 'Nilai tidak valid'
 
    time.sleep(1)
    print('Sedang diproses')
    time.sleep(0.5)
    dots = '...'
    for dot in dots:
        print(dot)
        time.sleep(0.5)
    print('Nilai dari matkul Anda masuk kategori ' + nilaihruf + '\n')
    time.sleep(1)
    ans = input("Apakah anda ingin mengulangi? (Y/T)")

print('\nProgram Berhenti')

File: UNIT 3/C/nilaiMatkulChecker.c
#include <stdio.h>
#include <string.h>

int main(){
    char nama[100], nim [20], matkul [100];
    char ans = 'Y', nilaihuruf;
    int nilai;
   while (ans == 'Y'|| ans == 'y') {
    printf("---Nilai Matkul Checker---\n");
    printf("Masukkan nama: ");
    scanf("%s", nama);
    printf("Masukkan NIM: ");
    scanf("%s", nim);
    printf("Masukkan mata kuliah: ");
    scanf("%s", matkul);
    printf("Masukkan nilai: ");
    scanf("%d", &nilai);
    printf("\nData berhasil diinput\n");

    if (nilai >= 85 && nilai <= 100){
        nilaihuruf = 'A';
    }else if (nilai >= 75 && nilai < 85){
        nilaihuruf = 'B';
    }else if (nilai >= 65 && nilai < 75){
        nilaihuruf = 'C';
    }else if (nilai < 65){
        nilaihuruf = 'D';
    }

    printf("\nNilai dari %c\n", nilaihuruf);
    printf("Apakah anda ingin mengulangi? (y/t): ");
    scanf(" %c", &ans);
   
    fflush(stdin);
   }
    return 0;
}